### PR TITLE
[OTE] Added deep-object-reid input parameters validation and tests

### PR DESCRIPTION
**Pull request by saltykox**
Added input parameters validations for functions and methods in external/deep-object-reid/torchreid_tasks directory

Task: CVS-80356

### CODE REVIEW

**Comment by saltykox** [comment]
ote-test/1512

File: ote_sdk/ote_sdk/utils/argument_checks.py [hunk 1]
```diff
@@ -0,0 +1,512 @@
...
+            for key, value in kwargs.items():
+                if key in input_parameters_values_map:
+                    raise TypeError(
+                        f"Duplication of the parameter {key} -- both in args and kwargs"
+                    )
+                input_parameters_values_map[key] = value
+            # Checking input parameters type
+            for parameter_name in expected_types_map:
+                parameter = input_parameters_values_map.get(parameter_name)
+                if parameter is None:
```
Comment (LeonidBeynenson) [inline ote_sdk/ote_sdk/utils/argument_checks.py]: Please, fix it
```suggestion
                if parameter_name not in input_parameters_values_map:
```
The reason: we cannot forbid to pass `None` value explicitly as a value of a parameter, e.g `a = dict(x=None, y=None)`

**Review by LeonidBeynenson — CHANGES_REQUESTED** [review]

File: ote_sdk/ote_sdk/utils/argument_checks.py (hunk 1 above)
Comment (saltykox) [reply ote_sdk/ote_sdk/utils/argument_checks.py]: thanks, I will fix it in https://github.com/openvinotoolkit/training_extensions/pull/891

**Comment by hlewando** [comment]
Tests:
http://validationreports.sclab.intel.com:8006/reports/build_number_report?test_session_build_number=ote-test-1748&environment=iotg-dev-workstation-12

**Comment by hlewando** [comment]
After merge tests failed:
http://validationreports.sclab.intel.com:8006/reports/build_number_report?test_session_build_number=ote-test-1751&environment=iotg-dev-workstation-12

Rebuild in progress:
https://ci-ote.iotg.sclab.intel.com/job/ote/job/ote-test/1759/

After rebase tests are stopped working due to having some issues with datasets.

**Comment by hlewando** [comment]
http://validationreports.sclab.intel.com:8006/reports/build_number_report?test_session_build_number=ote-test-1790&environment=iotg-dev-workstation-09

3 unittests are failed.

**Comment by hlewando** [comment]
@goodsong81 @sstrehlk  I think this part of code could be merged.

**Pull request merged**